types/scratch-vm/scratch-vm-tests.ts

```diff
@@ -0,0 +1,27 @@
+/// <reference types=".">
+
+class ExampleExtension implements ScratchExtension {
+    getInfo(): ExtensionMetadata {
+        return {
+            id: "HelloWorld",
+            name: "Hello World",
+            blocks: [{
+                opcode: "helloWorld",
+                blockType: "reporter",
+                text: "Return text [input]",
+                arguments: {
+                    input: {
+                        type: Scratch.extensions.ArgumentType.STRING,
+                        defaultValue: "Hello World"
+                    }
+                }
+            }]
+        };
+    }
+
+    helloWorld({ input }: { input: string }): string {
+        return input;
+    }
+}
+
+Scratch.extensions.register(new ExampleExtension());
```

Comment:
this could be wrong, so I'd check this out if used in tests, the extensions are not created directly.
Hence the requirement for single constructor parameter,
see implementation:
https://github.com/LLK/scratch-vm/blob/develop/src/extension-support/extension-manager.js#L132

Reply:
It turns out, you just linked to the extension loader for internal extensions. In the real world, that sort of thing doesn't exist.